### ISSUE TITLE
QoI has no access to instance being explained

### ISSUE THREAD

**Issue by piotrm0** [truera/trulens#72]
Quantities of interests receive output values on DoI samples but these are not necessarily the instance being explained, except in some cases like point. Because of this, one cannot define a QoI for "explained instance predicted class logits" because predicted class is not know unless the DoI sample is the same as the explained instance.

A related issue is that some may interpret QoI like the default max to mean what I described above but that is not what the max QoI implements because of the issue noted.
